### ISSUE TITLE
Accessing image environment variable via docker commit --change

### ISSUE THREAD

**Issue by IvanRibakov** [docker/cli#940]
**Description**

I am building Docker image using Packer. As such, I'm unable to use Dockerfile and have to use other tools for configuration management. As part of provisioning I am installing TexLive to `/usr/local/texlive/bin`. I need these folder to be on the PATH. Because I will be invoking executables inside container directly (without running bash first) I can't use `.bashrc` and other files that bash typically scans to update `$PATH`.

Since Packer allows me to leverage `docker commit` `--change` flag, I tried using it to update image's $PATH while committing it, however what I got was not what I expected. Depending on the notation used I would either end up passing host machine's `$PATH` value to the container or the only directory on the `$PATH` would be the one I'm adding.

As I was having doubts whether what I am trying to do is supported by `--change` flag I tried to look for confirmation of that in the documentation, but unfortunately nothing in the [docker commit](https://docs.docker.com/engine/reference/commandline/commit/#extended-description) or [Dockerfile ENV](https://docs.docker.com/engine/reference/builder/#environment-replacement) documentation made things any clearer for me so I'm still not 100% sure whether this is a bug or not. 

**Steps to reproduce the issue:**
1. `$ docker run -it centos:6 /bin/bash`
2. `[root@3b459018aa84 /]# echo $PATH`
    `/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin`
3. Get ID of the created container
4. Try one of the following:

```
    $ docker commit -c "ENV PATH /usr/local/texlive/bin:${PATH}" 3b459018aa84 test-changes
    $ docker commit -c 'ENV PATH /usr/local/texlive/bin:${PATH}' 3b459018aa84 test-changes
    $ docker commit -c "ENV PATH=\"/usr/local/texlive/bin:${PATH}\"" 3b459018aa84 test-changes
    $ docker commit -c 'ENV PATH=\"/usr/local/texlive/bin:${PATH}\"' 3b459018aa84 test-changes
    $ docker commit -c 'ENV PATH="/usr/local/texlive/bin:${PATH}"' 3b459018aa84 test-changes
    $ docker commit -c 'ENV PATH="/usr/local/texlive/bin:${PATH}"' -p=false 3b459018aa84 test-changes
```

4. `$ docker run -it test-changes /bin/bash`
6. `[root@a683ab06f4d3 /]# echo $PATH`

**Describe the results you received:**

`/usr/local/texlive/bin:` 
OR
`"/usr/local/texlive/bin:"`
OR `/usr/local/texlive/bin:/opt/local/bin:/opt/local/sbin:/Users/Ivan/.jenv/shims:/Users/Ivan/.jenv/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/usr/local/share/dotnet:/opt/X11/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands` (host machine $PATH) 
depending on the exact command used in step 4.

**Describe the results you expected:**

`/usr/local/texlive/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin` which is what I would get if I were to build the image using the following Dockerfile:

```
FROM centos:6

ENV PATH "/usr/local/texlive/bin:${PATH}"
```

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client:
 Version:	17.12.0-ce
 API version:	1.35
 Go version:	go1.9.2
 Git commit:	c97c6d6
 Built:	Wed Dec 27 20:10:14 2017
 OS/Arch:	linux/amd64

Server:
 Engine:
  Version:	17.12.0-ce
  API version:	1.35 (minimum version 1.12)
  Go version:	go1.9.2
  Git commit:	c97c6d6
  Built:	Wed Dec 27 20:12:46 2017
  OS/Arch:	linux/amd64
  Experimental:	false

```

**Output of `docker info`:**

```
Containers: 29
 Running: 0
 Paused: 0
 Stopped: 29
Images: 21
Server Version: 17.12.0-ce
Storage Driver: overlay2
 Backing Filesystem: xfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 89623f28b87a6004d4b785663257362d1658a729
runc version: b2567b37d7b75eb4cf325b77297b140ea686ce8f
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 3.10.0-693.2.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 15.4GiB
Name: localhost.localdomain
ID: YK54:X7YQ:OOWO:6FS3:KBJI:RY2V:7D5Z:CBCC:CAVE:C45D:EFTF:EN7P
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Username: <some username>
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by dnephin** [comment]
`docker commit -c 'ENV PATH /usr/local/texlive/bin:${PATH}' ` should be fixed by #35582, which I think should be in version 18.01.

Can you try updating Docker?

**Comment by IvanRibakov** [comment]
Yes, updating to Edge version solved the problem. Thank you for your help!
